src/zarr/core/array_spec.py

```diff
@@ -29,12 +29,11 @@ def __init__(
         prototype: BufferPrototype,
     ) -> None:
         shape_parsed = parse_shapelike(shape)
-        dtype_parsed = parse_dtype(dtype)
         fill_value_parsed = parse_fill_value(fill_value)
         order_parsed = parse_order(order)
 
         object.__setattr__(self, "shape", shape_parsed)
-        object.__setattr__(self, "dtype", dtype_parsed)
+        object.__setattr__(self, "dtype", dtype)
```

Comment:
i'm not sure about this -- the dtype is _annotated_ as `np.dtype[Any]` but we don't know if that's what we will get at runtime. I feel like parsing is safer, and more consistent with the rest of the` __init__` routine

Reply:
fair point. This PR introduces separate dtype parsers for v2 and v3. Maybe we can simply call `np.dtype` in this `__init__`?

Reply:
yeah, that also seems fine. we only use this class internally, and long term I envision a future in which this class disappears, but that might take a while.